src/components/Highlight/index.tsx

```diff
@@ -30,12 +31,29 @@ const Highlighted = styled.span`
  * Component will render a SPAN or series of SPAN with the content and highlights
  */
 export const Highlight: React.FC<HighlightProps> = ({ text, matches, highlight, ignoreCase = true, theme }) => {
-  if (!matches && !highlight) {
+  if (!matches && undefined === highlight) {
     throw new Error('You must set either indices or highlight');
   }
 
   if (matches) {
     let lastMatch = 0;
+    matches.forEach(match => {
+      if (
+        !Array.isArray(match) ||
```

Comment:
Could you extract this into the function `validateIndices(indices) { /*...*/ }`?
I would also throw for every case a respective message:

- [ ]  `(!Array.isArray(match) || match.length !== 2)` ->  `match must be an array of [start, end]`;
- [ ] `!Number.isInteger(match[0]) || !Number.isInteger(match[1])` -> `match must contain only positive integers`;
- [ ] `match[0] >= match[1]` -> `[start] must be lower than [end]`;
- [ ] `match[0] <= lastMatch` -> `the indices cannot overlap`.

Reply:
Still not fixed, right?
```tsx
const matches = "string"; // or {}, Date, function, etc.

<Highlight text="Hello World! Searching for World!" matches={matches} />

// TypeError: matches.map is not a function
```

Reply:
I saw the comment about the messages coming, but I thought it would pass :p, pushing new changes now